clients/java/signalr/src/main/java/com/microsoft/aspnet/signalr/HubConnection.java

```diff
@@ -185,9 +215,29 @@ public CompletableFuture start() throws Exception {
             return transport.send(handshake).thenRun(() -> {
                 hubConnectionStateLock.lock();
                 try {
+                    resetServerTimeout();
                     hubConnectionState = HubConnectionState.CONNECTED;
                     connectionState = new ConnectionState(this);
                     logger.log(LogLevel.Information, "HubConnected started.");
+
+                    this.pingTimer = new Timer();
+                    this.pingTimer.schedule(new TimerTask() {
+                        @Override
+                        public void run() {
+                            try {
+                                if (nextServerTimeout < System.currentTimeMillis()) {
+                                    stop("Server timeout elapsed without receiving a message from the server.");
```

Comment:
Do we also want to include the actual server timeout duration like the .NET Client?
https://github.com/aspnet/SignalR/blob/release/2.2/src/Microsoft.AspNetCore.SignalR.Client.Core/HubConnection.cs#L932